docs/gethue/content/en/posts/2021-05-26-importer-direct-upload.md

```diff
@@ -0,0 +1,94 @@
+  - light
+sf_background_image_size:
+  - cover
+sf_social_sharing:
+  - 1
+sf_related_articles:
+  - 1
+sf_sidebar_config:
+  - left-sidebar
+sf_left_sidebar:
+  - Sidebar-2
+sf_right_sidebar:
+  - Sidebar-1
+sf_caption_position:
+  - caption-right
+sf_remove_promo_bar:
+  - 1
+ampforwp-amp-on-off:
+  - default
+categories:
+  - Version 4.10
+  - Development
+
+---
+
+If you’ve ever struggled with creating new SQL tables from CSV files, you’ll be happy to learn that this is now much easier.
+
+If you know the history of Hue then you must be familiar with the importer option in Hue. So basically earlier we could create a table from a file in different databases but there is one necessary condition, the file must be present in the filesystem storage like [HDFS](https://hadoop.apache.org/docs/r1.2.1/hdfs_design.html#Introduction) or [s3](https://gethue.com/introducing-s3-support-in-hue/). Now we can create a table from local storage files also in different [databases](https://docs.gethue.com/administrator/configuration/connectors/).
+
+### Goal:
+
+* Propose a direct upload of small file which does not depend on a storage system in the importer.
+
+### Why:
+
+* If someone doesn't have HDFS or s3 configured in Hue then it is difficult to create the table from a file, so if we have the feature like this then anyone can use this feature for creation of tables. And ofcourse, it also saves time as it directly uploads the file to the importer.
+
+### Steps to create a table:
+
+![Importer direct upload steps gif](https://cdn.gethue.com/uploads/2021/05/Importer_direct_upload_steps.gif)
+
+### Describing the steps through the app diagram:
+![Importer direct upload steps app diagram](https://cdn.gethue.com/uploads/2021/05/Importer_direct_upload_app_diagram.png)
+
+### Files and APIs:
+
+  * We are using 3 APIs to implement this feature.
+    * [Guess_format](https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/api3.py#L121) (to guess the file format)
+    * [Guess_field_types](https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/api3.py#L228) (to guess the column types)
+    * [Importer_submit](https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/api3.py#L444) (to create a table)
+  * You can look at [sql.py](https://github.com/cloudera/hue/blob/master/desktop/libs/indexer/src/indexer/indexers/sql.py) file If you are more curious about how we are dealing with different types of dialects or sql.
+
+### Intermediate steps(ss):
+![Importer direct upload step1](https://cdn.gethue.com/uploads/2021/05/Importer_direct_upload_step1.png)
+</br>
+  Selection of file in (small local file), Format guess and preview of table.
+</br>
+
+![Importer direct upload step2](https://cdn.gethue.com/uploads/2021/05/importer_direct_upload_step2.png)
+</br>
+  Selection of dialect and auto field type guesses. You can edit the name and type of fields by clicking at the Fields section.
```

Comment:
Suggested edit: You select a SQL dialect and Hue auto-detects the column data types. You can edit column names and their data type.